code-challenges.test.js

```diff
@@ -1,59 +1,223 @@
+
+// I am getting lost in the function itself.
+
+
+// const randomNouns = randomNouns1.map => {
+//     return randomNouns1.toUpperCase()
+// }
+
+// //   output: Test Suites: 1 failed, 1 total
+
+
+// const randomNouns = array.map(array); {
+//     return array(array.toUpperCase())
+// }
+
+// Whatever we are mapping over needs to be the arguement we are mapping through
+// randomNouns1.((string)) => {
+
+    // return randomNouns1(randomNouns1.toUpperCase) }
+   
+    //    return (randomNouns.toUpperCase())
+
+
+
+
+// //   ReferenceError: randomNouns is not defined
+
+// //       105 |         const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+// //       106 |         const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+// //     > 107 |       expect(randomNouns(randomNouns1)).toUpperCase("STREETLAMP", "POTATO", "TEETH", "CONCLUSION", "NEPHEW")
+//           |       ^
+//       108 |       expect(randomNouns(randomNouns2)).toUpperCase("TEMPERATURE", "DATABASE", "CHOPSTICKS", "MAMGO")
+//       109 |     })
+//       110 |   })
+
+
+
+
+// // --------------------3) 
+
+// Create a function that takes in a string and logs the index of the first vowel.
+
+// // a) Create a test with expect statements for each of the variables provided.
+
+// const vowelTester1 = "learn"
+// // // Expected output: 1
+// const vowelTester2 = "academy"
+// // // Expected output: 0
+// const vowelTester3 = "challenges"
+// // // Expected output: 2
+
+// // // b) Create the function that makes the test pass.
+
+
+// describe("vowel", () => {
+//     it("logs the index of the first vowel", () => {
+//         const vowelTester1 = "learn"
+//         const vowelTester2 = "academy"
+//         const vowelTester3 = "challenges"
+//       expect(vowel(vowelTester1)).indexOf("1") 
+//       expect(vowel(vowelTester2)).indexOf("0")
+//       expect(vowel(vowelTester3)).indexOf("2")
+//     })
+//   })
+
+// //   ReferenceError: vowel is not defined
+
+// const vowel =(string) =>{
+//     // return string.indexOf("learn"[1])
+//     let count = 0
+// for(let i = 0; i < string.length; i++) {
+//   if(string[i] === "a","e","i","o","u") {
+//     count++
+//   } return {[i]string
 
-// b) Create the function that makes the test pass.
+//   }
+// }
+// }
 
-// --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
-// a) Create a test with expect statements for each of the variables provided.
+//   const string = (vowel) => {
+//     if(string[value] === "a","e","i","o","u") {count++
 
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
-const vowelTester3 = "challenges"
-// Expected output: 2
+// Right here I started off by creating a new variable with a string
+// I used indexeOf beause it will take a value as an arguement and return the first index that it finds that has that data
```

Comment:
I am not seeing a lot of pseudocode to help break down the problem into smaller pieces.  As these problems become more complex, the more crucial it is going to be to pseudocode the logic before making attempts at writing the code.  Also, make sure to paste any error codes you are getting when running your test.